schema/interface_definition.rnc

```diff
@@ -184,3 +187,15 @@ completionHelp = element completionHelp
     (element path { text })* &
     (element script { text })*
 }
+
+
+# childRequirements tags is a declarative way to configure basic
+# requirements of node or tagnode children.
+childRequirements = element childRequirements {
+    (element require { child+ } )? &
+    (element conflict { nodeNameAttr, child+ })* &
+    (element atLeastOneOf { child+ } )* &
+    (element depend { nodeNameAttr, child+ })*
```

Comment:
I'm not sure if we should use both `require` and `depend`, from the maintainability perspective.

Reply:
> I'm not sure if we should use both `require` and `depend`, from the maintainability perspective.

I assume what you mean here is that since both fields can be intepreted as "this node must have these children configured" it can be confusing when not taken with extra context, which will waste mental bandwidth.

If that is the case what about `(element requiredChildren { child+ } )? &` and `(element childDependsOn { nodeNameAttr, child+ })*` instead?